lib/pengine/complex.c

```diff
@@ -596,16 +596,15 @@ pe__unpack_resource(xmlNode *xml_obj, pe_resource_t **rsc,
 
     if (id == NULL) {
         pe_err("Must specify id tag in <resource>");
-        return FALSE;
+        return pcmk_rc_schema_validation;
```

Comment:
The common existing use cases of `pcmk_rc_schema_validation` throughout Pacemaker are much like this one: return this error if something about the XML is invalid or doesn't align with our expectations, like a missing attribute or an attribute that refers to something it shouldn't.

With that said, it doesn't align with the documented use case of `pcmk_rc_schema_validation`: `"Update does not conform to the configured schema"`. Eventually, it would be nice to either clean these up or make the message more inclusive. For now, I'm using this RC the way it's often used elsewhere.

Reply:
I think your new code is better throughout

Reply:
I'm taking this to mean use `pcmk_rc_unpack_error` for all five error-returns.

Reply:
yep

Reply:
and for a lot of the preexisting uses